cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1244,12 +1244,12 @@ class ObjectOutputStream final : public io::OutputStream {
 
     if (current_part_) {
       // Upload last part
-      RETURN_NOT_OK(CommitCurrentPart());
+      RETURN_NOT_OK(CommitCurrentPart(true));
```

Comment:
Add a comment to make it easier to reason the boolean argument at caller side?
`CommitCurrentPart(/*always_foreground_writes=*/true)`

Reply:
Sure, I've gotten away from the practice of adding the extra comments as my IDE does this for me, but done.